sdk/cosmos/tests.yml

```diff
@@ -0,0 +1,429 @@
+trigger:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - sdk/cosmos
+
+variables:
+  pomFile: './sdk/cosmos/pom.xml'
+  DefaultOptions: '--batch-mode -Dmaven.wagon.http.pool=false'
+  LoggingOptions: '-Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+
+jobs:
+  - job: 'Build'
+
+    pool:
+      vmImage: 'vs2017-win2016'
+
+    steps:
+    - task: Maven@3
+      displayName: 'Install reporting tools'
+      inputs:
+        mavenPomFile: ./eng/code-quality-reports/pom.xml
+        options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
+        mavenOptions: '$(LoggingOptions)'
+        javaHomeOption: 'JDKVersion'
+        jdkVersionOption: '1.11'
+        jdkArchitectureOption: 'x64'
+        publishJUnitResults: false
+        goals: 'install'
+
+    - task: Maven@1
+      displayName: Build
+      inputs:
+        mavenPomFile: $(pomFile)
+        goals: 'package package'
+        options: '-DskipTests -Dgpg.skip'
+        publishJUnitResults: false
+        checkStyleRunAnalysis: true
+        pmdRunAnalysis: true
+        findBugsRunAnalysis: true
+      condition: succeededOrFailed()
+
+  - job: 'Test'
+
+    timeoutInMinutes: 240
+
+    strategy:
+      maxParallel: 1
+      matrix:
+        Session_Single_Region_ReadMyWrites:
+          OSVmImage: 'vs2017-win2016'
+          CosmosAccountHost: $(SINGLE_SESSION_ONE_ACCOUNT_HOST)
+          CosmosAccountKey: $(SINGLE_SESSION_ONE_ACCOUNT_KEY)
```

Comment:
Secrets should be stored in accordance with our testing guidelines.

Reply:
updated to use variable groups.